gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/execute/TablePrivilegeInfo.java

```diff
@@ -229,10 +230,11 @@ public void executeGrantRevoke( Activation activation,
 	{
 		doExecuteGrantRevoke(activation, grant, grantees, columnBitSets, actionAllowed, true, td);
 		GemFireStore ms = Misc.getMemStore();
-		if (ms.isSnappyStore() && Misc.isSecurityEnabled()
-			&& ms.getExternalCatalog().isColumnTable(td.getSchemaName(), td.getName(), true)) {
-			String cbTable = CallbackFactoryProvider.getStoreCallbacks().columnBatchTableName(td
-					.getName());
+		ExternalCatalog ec = ms.getExternalCatalog(); // This may be null during restart
+		String cbTable = CallbackFactoryProvider.getStoreCallbacks().columnBatchTableName(td
+				.getName());
+		if (ms.isSnappyStore() && Misc.isSecurityEnabled() && ((ec != null && ec.isColumnTable(td
+				.getSchemaName(), td.getName(), true)) || Misc.getRegion(cbTable,false, true) != null)) {
```

Comment:
If ec is null we can put an assertion that ddlreplayinprogress should be true.

Please check initialDDLReplayInProgress. If it is false throw IllegalStateException.